### ISSUE TITLE
Feature Request: Show details of document and edit them

### ISSUE THREAD

**Issue by totti4ever** [bauerj/paperless_app#5]
Currently I can only open the file when I click on it. Editing would be extremely helpful especially for tags, to go through them when being in the bus or in bed or so

**Comment by bauerj** [comment]
I agree, this could be very useful.

I will look into this after implementing the offline mode.

**Comment by mvdkleijn** [comment]
I'd suggest splitting this into two features: one showing the details and one allowing for edits...

**Comment by Akshay0701** [comment]
yes @mvdkleijn it will be simple way

**Comment by bauerj** [comment]
Hey,

this has now been implemented:

![image](https://user-images.githubusercontent.com/598790/141992694-548f959c-359e-45f5-83bb-27b9dd74fc41.png)

Editing is possible as well (by tapping on the small pencil buttons). Documents can be renamed and deleted through the context menu.

I will release an update in 2 weeks or so. Extensive code changes have been necessary to maintain compatibility with latest Flutter SDK. It would be nice if interested users could beta-test this until then to increase chances of spotting any remaining bugs.

You can download a development build here: 
[app-armeabi-v7a-release.zip](https://github.com/bauerj/paperless_app/files/7546660/app-armeabi-v7a-release.zip)
 

Please note that you will have to uninstall the app first before being able to install the development build.

**Comment by mvdkleijn** [comment]
I will try to beta this coming friday when I should have some time

**Comment by mvdkleijn** [comment]
> I will try to beta this coming friday when I should have some time

OK, this is a little weird. It seems to be refusing to (try to) authenticate. The version from the playstore authenticates just fine though.

**Comment by mvdkleijn** [comment]
@bauerj For the life of me, I cannot get this build to log into my paperless-ng. The version on the play store works fine for me but this beta doesn't.

It installs and starts just fine and I can enter the server details without it complaining but when I try to login using my credentials, it just sits there and does nothing.

So in essence: I was unable to test, sorry.

**Comment by bauerj** [comment]
Sorry for responding so late! I just tested the APK and it doesn't work for me either 🤦‍♂️

I'll see if I can find out what's wrong and will get back to you then.

**Comment by mvdkleijn** [comment]
Ever found out what the problem was? :smile:

**Comment by bauerj** [comment]
Yes, I was able to fix the issue (it was related to the null safety transition) but I was in the middle of improving the search bar so that's kind of broken right now.

I hope I can send you a working APK after the holidays.

**Comment by bauerj** [comment]
Hey @mvdkleijn,

I have a new version that I'm able to run on my local device:

[app-armeabi-v7a-release.zip](https://github.com/bauerj/paperless_app/files/7785564/app-armeabi-v7a-release.zip)

Please let me know how it works out for you.

There are a few more things to test if you'd like:

* You can now filter for tags as well as correspondents through the search bar
* Documents can be edited

**Comment by mvdkleijn** [comment]
@bauerj Sorry, I'm having some weird behavior. Its almost like some stuff like pwd responses are being cached despite me flushing the app from my phone. I'll give it another go later. Maybe @totti4ever has more luck testing

**Comment by qcasey** [comment]
Most of these new features work great for me. Filtering by correspondent works perfectly as well, but filtering by tag does not.

Tapping a tag name seems to have no effect on the results. I can `flutter run` this branch and pull what's happening tomorrow if that helps

**Comment by qcasey** [comment]
Okay I fixed the tag filter in https://github.com/bauerj/paperless_app/commit/39e0ede9c67e2bda5fe71c8ae195764990a0a7ae on my device. Mind taking a look to see if it works for you as well?

Also I believe the search bar needed some contrast (especially on the light theme), which I added in https://github.com/bauerj/paperless_app/commit/4b3c2294af76b94abf1a76f0f5c230259bab87bd. Just made the text white and removed the icon. What do you think?

![Screenshot_20220124-135613_Paperless~2](https://user-images.githubusercontent.com/1429672/150872234-e6c15221-e4b3-439d-a9e7-c52fd7e855ed.png)
![Screenshot_20220124-135543_Paperless~2](https://user-images.githubusercontent.com/1429672/150872240-4aa1cd9f-e381-4a86-b079-5ef89bb71b2b.png)

**Comment by bauerj** [comment]
It works (and looks) great, thanks!

**Comment by bauerj** [comment]
I just released a new version to public testing and F-Droid. If nobody complains until February 7, I will release it and close this issue.

**Comment by bauerj** [comment]
There have been some complaints but I think most bugs are fixed now.
